connectors/src/connectors/github/index.ts

```diff
@@ -459,6 +462,218 @@ export async function retrieveGithubReposTitles(
+  }
+
+  const githubInstallationId = c.connectionId;
+  const allReposIdsToFetch: Set<number> = new Set();
+  const nodes: ConnectorNode[] = [];
+
+  // Users can select:
+  // A full repo (issues + discussions + code if enabled)
+  const fullRepoIds: number[] = [];
+
+  //  All issues of the repo, or all discussions of the repo
+  const allIssuesFromRepoIds: number[] = [];
+  const allDiscussionsFromRepoIds: number[] = [];
+
+  // The full code, or a specific folder or file in the code
+  const allCodeFromRepoIds: string[] = [];
+  const codeDirectoryIds: string[] = [];
+  const codeFileIds: string[] = [];
+
+  // We loop on all the internalIds we receive to know what is the related data type
+  internalIds.forEach((internalId) => {
+    const { type, repoId } = matchGithubInternalIdType(internalId);
+    allReposIdsToFetch.add(repoId);
+
+    switch (type) {
+      case "REPO_FULL":
+        fullRepoIds.push(repoId);
+        break;
+      case "REPO_ISSUES":
+        allIssuesFromRepoIds.push(repoId);
+        break;
+      case "REPO_DISCUSSIONS":
+        allDiscussionsFromRepoIds.push(repoId);
+        break;
+      case "REPO_CODE":
+        allCodeFromRepoIds.push(repoId.toString());
+        break;
+      case "REPO_CODE_DIR":
+        codeDirectoryIds.push(internalId);
+        break;
+      case "REPO_CODE_FILE":
+        codeFileIds.push(internalId);
+        break;
+      default:
+        throw new Error(`Invalid type: ${type}`);
+    }
+  });
+
+  // Repos are not stored in the DB, we have to fetch them from the API
+  const uniqueRepoIdsArray: number[] = Array.from(allReposIdsToFetch);
+  const uniqueRepos: Record<number, GithubRepo> = {};
+  await concurrentExecutor(
+    uniqueRepoIdsArray,
+    async (repoId) => {
+      const repoData = await getRepo(githubInstallationId, repoId);
+      uniqueRepos[repoId] = repoData;
+    },
+    { concurrency: 8 }
+  );
+
+  // Code Repositories, Directories and Files are stored in the DB
+  const [fullCodeInRepos, codeDirectories, codeFiles] = await Promise.all([
+    GithubCodeRepository.findAll({
+      where: {
+        connectorId: c.id,
+        repoId: allCodeFromRepoIds,
+      },
+    }),
+    GithubCodeDirectory.findAll({
+      where: {
+        connectorId: c.id,
+        internalId: codeDirectoryIds,
+      },
+    }),
+    GithubCodeFile.findAll({
+      where: {
+        connectorId: c.id,
+        documentId: codeFileIds,
+      },
+    }),
+  ]);
+
+  // Constructing Nodes for Full Repo
+  fullRepoIds.forEach((repoId) => {
+    const repo = uniqueRepos[repoId];
+    if (!repo) {
+      return;
+    }
+    nodes.push({
+      provider: c.type,
+      internalId: repoId.toString(),
+      parentInternalId: null,
+      type: "folder",
+      title: repo.name,
+      titleWithParentsContext: `[${repo.name}] - Full repository`,
+      sourceUrl: repo.url,
+      expandable: true,
+      permission: "read" as ConnectorPermission,
```

Comment:
can you change this to not have `as ConnectorPermission` ? with the current code, if we ever change ConnectorPermission such that "read" is no longer valid, we won't get the type error here

Reply:
Done in all my function + on the code for getPermissions that was using the same!

Reply:
Awesome thank you !